asf_tools/composite.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+"""Create S1 SAR Composite Mosaic using inverse area weighting ala David Small.
+
+   Path vs infiles:
+     If path is passed, code assumes files are in an ASF HyP3 RTC Stacking arrangement.
+     i.e  {path}/20*/PRODUCT/ contains the input RTC data and the area maps or
+          {path}/S1?_IW_*RTC*/ contains the input RTC data and the area maps
+         
+
+"""
+
+import numpy as np
+import logging
+import os
+import re
+import argparse
+import glob
+# from hyp3lib import saa_func_lib as saa
+# import hyp3lib.saa_func_lib as saa
+import saa_func_lib as saa
+from osgeo import gdal
+from osgeo.gdalconst import GRIORA_Cubic
+
+
+def get_pol(infile):
+    """Return the polarization of infile"""
+    if "VV" in infile:
+        pol = "VV"
+    elif "VH" in infile:
+        pol = "VH"
+    elif "HH" in infile:
+        pol = "HH"
+    elif "HV" in infile:
+        pol = "HV"
+    else:
+        raise Exception("Could not determine polarization of file " + infile)  
+    return pol
+
+
+def frange(start, stop=None, step=None):
+    """Return a floating point number ranging from start to stop, adding step"""
+    if not stop:
+        stop = start + 0.0
+        start = 0.0
+    if not step:
+        step = 1.0
+    while True:
+        if step > 0 and start >= stop:
+            break
+        elif step < 0 and start <= stop:
+            break
+        yield ("%g" % start)  # return float number
+        start = start + step
+
+
+def get_full_extent(corners):
```

Comment:
All these spacial helper functions should probably be moved to `asf_tools.geometry` (or similar) and tested